### ISSUE TITLE
Incorrect identification of sqlite keywords

### ISSUE THREAD

**Issue by Ttibsi** [sqlfluff/sqlfluff#4398]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```sql
CREATE TABLE users (
    user_id INTEGER PRIMARY KEY AUTOINCREMENT,
    username TEXT NOT NULL UNIQUE,
    password TEXT NOT NULL,
    email TEXT NOT NULL UNIQUE
);
```

This is returning "L029 Keywords should not be used as identifiers" in line 4 pos 5 (the `password` name). This is valid sqlite, and isn't listed as a keyword here: https://www.sqlite.org/lang_keywords.html. Removing the "UNIQUE" on the previous line (to check if it was incorrectly pointing there instead) changed nothing. Playing around in the online generator shows that it specifically doesn't like "password" as a column name. 

### Expected Behaviour

No errors

### Observed Behaviour

L029 | 4 / 5 | Keywords should not be used as identifiers.

### How to reproduce

```sql
CREATE TABLE users (
    user_id INTEGER PRIMARY KEY AUTOINCREMENT,
    username TEXT NOT NULL,
    password TEXT NOT NULL,
    email TEXT NOT NULL UNIQUE
);
```

### Dialect

sqlite

### Version

online.sqlfluff.com

### Configuration

na

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by pdebelak** [comment]
`password` is listed as an "unreserved keyword" for the sqlite dialect which means that this is valid SQL but rule L029 still thinks you shouldn't use it as an identifier as a style suggestion.

> Although 'unreserved' keywords 'can' be used as identifiers, and 'reserved words' can be used as quoted identifiers, best practice is to avoid where possible, to avoid any misunderstandings as to what the alias represents.

Personally, I add L029 to the `exclude_rules` list in my configuration since it flags things that are difficult to avoid or that I don't care about.
